rts/at/integration/src/main/java/org/jboss/narayana/rest/integration/RecoveryManager.java

```diff
@@ -96,60 +95,75 @@ public void persistParticipantInformation(final ParticipantInformation participa
-            } catch (ObjectStoreException ose) {
-                RESTATLogger.atI18NLogger.warn_failureRemovingParticipantObjectStore("Failure while removing participant information from the object store.", ose);
+            } catch (Exception e) {
+                String msg = RESTATLogger.atI18NLogger.warn_persistParticipantInformationRecoveryManager(e.getMessage(), e);
+                throw new ParticipantException(msg, e);
             }
         }
     }
 
     private OutputObjectState getParticipantInformationOutputState(final ParticipantInformation participantInformation)
-            throws IOException {
-
-        final Uid uid = new Uid(participantInformation.getId());
-        final OutputObjectState state = new OutputObjectState(uid, PARTICIPANT_INFORMATION_RECORD_TYPE);
+            throws ParticipantException {
+        try {
+            final Uid uid = new Uid(participantInformation.getId());
+            final OutputObjectState state = new OutputObjectState(uid, PARTICIPANT_INFORMATION_RECORD_TYPE);
 
-        state.packString(participantInformation.getId());
-        state.packString(participantInformation.getApplicationId());
-        state.packString(participantInformation.getStatus());
-        state.packString(participantInformation.getRecoveryURL());
-        state.packBytes(getParticipantBytes(participantInformation.getParticipant()));
+            state.packString(participantInformation.getId());
+            state.packString(participantInformation.getApplicationId());
+            state.packString(participantInformation.getStatus());
+            state.packString(participantInformation.getRecoveryURL());
+            state.packBytes(getParticipantBytes(participantInformation.getParticipant()));
 
-        return state;
+            return state;
+        } catch (Exception e) {
+            String msg = RESTATLogger.atI18NLogger.warn_persistParticipantInformationRecoveryManager(e.getMessage(), e);
+            throw new ParticipantException(msg, e);
+        }
     }
 
-    private byte[] getParticipantBytes(final Participant participant) throws IOException {
-        if (participant instanceof Serializable) {
-            return serializeParticipant((Serializable) participant);
-        } else if (participant instanceof PersistableParticipant) {
-            return ((PersistableParticipant) participant).getRecoveryState();
+        private byte[] getParticipantBytes(final Participant participant) throws ParticipantException {
+        try {
+            if (participant instanceof Serializable) {
+                return serializeParticipant((Serializable) participant);
+            } else if (participant instanceof PersistableParticipant) {
+                return ((PersistableParticipant) participant).getRecoveryState();
+            }
+        } catch (Exception e) {
+            String msg = RESTATLogger.atI18NLogger.warn_persistParticipantInformationRecoveryManager(e.getMessage(), e);
```

Comment:
The code in this catch block is identical to the catch block in getParticipantInformationOutputState on line 135 which is the only caller. I would recommend that you don't catch any exception and let getParticipantInformationOutputState handle it.